tests/testthat/test-get_code.R

```diff
@@ -0,0 +1,476 @@
+warning_message <- "warning('Code was not verified for reproducibility.')"
+
+testthat::test_that("get_code with datanames does not brake for empty code", {
+  testthat::expect_identical(
+    get_code(teal_data(a = 1, code = character(0)), datanames = "a"),
+    warning_message
+  )
+  testthat::expect_identical(
+    get_code(teal_data(a = 1, code = ""), datanames = "a"),
+    c(warning_message, "")
+  )
+})
+
+testthat::test_that("get_code with datanames extracts code of a binding from a simple code put in a character", {
+  code <- c(
+    "a <- 1",
+    "b <- 2"
+  )
+  tdata <- eval_code(teal_data(), code)
+  datanames(tdata) <- c("a", "b")
+  testthat::expect_identical(
+    get_code(tdata, datanames = "a"),
+    "a <- 1"
+  )
+  testthat::expect_identical(
+    get_code(tdata, datanames = "b"),
+    "b <- 2"
+  )
+})
+
+testthat::test_that("get_code with datanames warns if binding doesn't exist in a code", {
+  code <- c(
+    "a <- 1",
+    "b <- 2"
+  )
+  tdata <- eval_code(teal_data(), code)
+  datanames(tdata) <- c("a", "b")
+  testthat::expect_error(
+    get_code(tdata, datanames = "c"),
+    "Assertion on 'datanames' failed"
+  )
+})
+
+
+testthat::test_that(
+  "get_code with datanames extracts code of a parent binding but only those evaluated before coocurence",
+  {
+    code <- c(
+      "a <- 1",
+      "b <- a",
+      "a <- 2"
+    )
+    tdata <- eval_code(teal_data(), code)
+    datanames(tdata) <- c("a", "b")
+    testthat::expect_identical(
+      get_code(tdata, datanames = "b"),
+      c("a <- 1", "b <- a")
+    )
+  }
+)
+
+testthat::test_that("get_code with datanames extracts code of a parent binding if used as an arg in fun call", {
+  code <- c(
+    "a <- 1",
+    "b <- identity(x = a)",
+    "a <- 2"
+  )
+  tdata <- eval_code(teal_data(), code)
+  datanames(tdata) <- c("a", "b")
+  testthat::expect_identical(
+    get_code(tdata, datanames = "b"),
+    c("a <- 1", "b <- identity(x = a)")
+  )
+})
+
+testthat::test_that("get_code with datanames is possible to output the code for multiple objects", {
+  code <- c(
+    "a <- 1",
+    "b <- 2",
+    "c <- 3"
+  )
+  tdata <- eval_code(teal_data(), code)
+  datanames(tdata) <- c("a", "b", "c")
+  testthat::expect_identical(
+    get_code(tdata, datanames = c("a", "b")),
+    c("a <- 1", "b <- 2")
+  )
+})
+
+testthat::test_that("get_code with datanames can't extract the code for assign function", {
+
+  testthat::skip("We will tackle 'assign' some day!")
+  code <- c(
+    "a <- 1",
+    "assign('b', 5)",
+    "b <- b + 2"
+  )
+  tdata <- eval_code(teal_data(), code)
+  datanames(tdata) <- c("a", "b")
+  testthat::expect_error(
+    get_code(tdata, datanames = "b"),
+    "b <- b + 2"
+  )
```

Comment:
It won't throw error if we support this. Suggested change is an expectation
```suggestion
  testthat::expect_identical(
    get_code(tdata, datanames = "b"),
    c("assign('b', 5)", "b <- b + 2")
  )
```